### PR TITLE
Add support for MongoDB testing module

### PR DESCRIPTION
**Pull request by ravilushqa**
This commit introduces functionalities for MongoDB testing, achieving 3 main goals:

1. Added mongodb module, `github.com/testcontainers/testcontainers-go/modules/mongodb`, in `dependabot.yml` with a monthly update cycle.

2. Created new `.md` documentation file for this module, providing details on how to add MongoDB to the Go dependencies, how to use this module, the MongoDB module reference, and other related details.

3. Implemented test files to verify MongoDB connection and integration, and added a `Makefile` for test execution.

Justification:
This helps users in testing their mongodb related go functionalities, effectively and efficiently.

## Related issues

-

### CODE REVIEW

**Comment by mdelapenya** [comment]
Could you remove in this PR the MongoDB example, located in `./examples/mongodb`?

**Comment by ravilushqa** [comment]
> Could you remove in this PR the MongoDB example, located in `./examples/mongodb`?

sure :)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -82,7 +82,7 @@ jobs:
       matrix:
         go-version: [1.19.x, 1.x]
         platform: [ubuntu-latest, macos-latest]
-        module: [compose, couchbase, k3s, localstack, mysql, neo4j, postgres, pulsar, redis, redpanda, vault]
+        module: [compose, couchbase, k3s, localstack, mongodb, mysql, neo4j, postgres, pulsar, redis, redpanda, vault]
```
Comment (mdelapenya) [inline .github/workflows/ci.yml:85]: Could you update https://github.com/testcontainers/testcontainers-go/pull/1447/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR103 consequently?

File: .github/dependabot.yml [hunk 2]
```diff
@@ -134,6 +127,12 @@ updates:
         day: sunday
       open-pull-requests-limit: 3
       rebase-strategy: disabled
+    - package-ecosystem: gomod
+      directory: /modules/mongodb
+      schedule:
+        interval: monthly
```
Comment (mdelapenya) [inline .github/dependabot.yml:133]: Forgot to add it 🤦 
```suggestion
        interval: monthly
        day: sunday
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ravilushqa) [reply .github/workflows/ci.yml:85]: done!

File: docs/modules/mongodb.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+- `testcontainers.WithConfigModifier`
+- `testcontainers.WithHostConfigModifier`
+- `testcontainers.WithEndpointSettingsModifier`
+
+Please read the [Create containers: Advanced Settings](../features/creating_container.md#advanced-settings) documentation for more information.
+
+### Container Methods
+
+The MongoDB container exposes the following methods:
```
Comment (mdelapenya) [inline docs/modules/mongodb.md:69]: Last thing to merge this PR: could you please add here the container method added, i.e. the ConnectionString? 🙏 You could use MySQL as reference: https://golang.testcontainers.org/modules/mysql/#connectionstring

I think once we have this in place, we are good to go! 🚀

Comment (ravilushqa) [reply docs/modules/mongodb.md:69]: Added. Please review the additional content to ensure its accuracy and completeness. Let me know if any further additions are required.

Comment (mdelapenya) [reply docs/modules/mongodb.md:69]: I code snippet would be much appreciated 🙏

File: docs/modules/mongodb.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+Please run the following command to add the MongoDB module to your Go dependencies:
+
+```
+go get github.com/testcontainers/testcontainers-go/modules/mongodb
+```
+
+## Usage example
+
+<!--codeinclude-->
+[Creating a MongoDB container](../../modules/mongodb/mongodb.go)
```
Comment (mdelapenya) [inline docs/modules/mongodb.md:20]: Do we want to share the entire file or just the blocks related to the creation of the container?

File: docs/modules/mongodb.md [hunk 5]
```diff
@@ -17,7 +17,7 @@ go get github.com/testcontainers/testcontainers-go/modules/mongodb
 ## Usage example
 
 <!--codeinclude-->
-[Creating a MongoDB container](../../modules/mongodb/mongodb.go) inside_block:createMongoDBContainer
+[Creating a MongoDB container](../../modules/mongodb/mongodb_test.go) inside_block:createMongoDBContainer
 <!--/codeinclude-->
 
 <!--codeinclude-->
```
Comment (mdelapenya) [inline docs/modules/mongodb.md:23]: We probably do not want this second tab here

**Review by mdelapenya — DISMISSED** [review]
LGTM! Will merge it right after the CI passes. Thank you so much!

**Comment by mdelapenya** [comment]
Because the mongo workflow passed, I'm merging this one now.

@ravilushqa thank you so much for converting the example into a module. I'm going to update the existing conversion guide, given we changed the CI workflow.

image

**Pull request merged**